runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -573,12 +574,331 @@ static TR::Instruction *generatePrefetchAfterHeaderAccess(TR::Node
+
+   TR::Node *child = node->getFirstChild();
+
+   // TODO: Perform f2l using SSE.
+   if (child->getOpCode().isDouble())
+      {
+      TR::RegisterDependencyConditions  *deps;
+
+      TR::Register        *doubleReg = cg->evaluate(child);
+      TR::Register        *lowReg    = cg->allocateRegister(TR_GPR);
+      TR::Register        *highReg   = cg->allocateRegister(TR_GPR);
+      TR::RealRegister    *espReal   = cg->machine()->getRealRegister(TR::RealRegister::esp);
+
+      deps = generateRegisterDependencyConditions((uint8_t) 0, 3, cg);
+      deps->addPostCondition(lowReg, TR::RealRegister::NoReg, cg);
+      deps->addPostCondition(highReg, TR::RealRegister::NoReg, cg);
+      deps->addPostCondition(doubleReg, TR::RealRegister::NoReg, cg);
+      deps->stopAddingConditions();
+
+      TR::LabelSymbol *reStartLabel = TR::LabelSymbol::create(cg->trHeapMemory(),cg);   // exit routine label
+      TR::LabelSymbol *CallLabel    = TR::LabelSymbol::create(cg->trHeapMemory(),cg);   // label where long (64-bit) conversion will start
+      TR::LabelSymbol *StartLabel   = TR::LabelSymbol::create(cg->trHeapMemory(),cg);
+
+      StartLabel->setStartInternalControlFlow();
+      reStartLabel->setEndInternalControlFlow();
+
+      // Attempt to convert a double in an XMM register to an integer using CVTTSD2SI.
+      // If the conversion succeeds, put the integer in lowReg and sign-extend it to highReg.
+      // If the conversion fails (the double is too large), call the helper.
+      generateRegRegInstruction(CVTTSD2SIReg4Reg, node, lowReg, doubleReg, cg);
+      generateRegImmInstruction(CMP4RegImm4, node, lowReg, 0x80000000, cg);
+
+      generateLabelInstruction(LABEL, node, StartLabel, cg);
+      generateLabelInstruction(JE4, node, CallLabel, cg);
+
+      generateRegRegInstruction(MOV4RegReg, node, highReg ,lowReg, cg);
+      generateRegImmInstruction(SAR4RegImm1, node, highReg , 31, cg);
+
+      generateLabelInstruction(LABEL, node, reStartLabel, deps, cg);
+
+      TR::Register *targetRegister = cg->allocateRegisterPair(lowReg, highReg);
+      TR::SymbolReference *d2l = comp->getSymRefTab()->findOrCreateRuntimeHelper(TR_IA32double2LongSSE);
+      d2l->getSymbol()->getMethodSymbol()->setLinkage(TR_Helper);
+      TR::Node::recreate(node, TR::lcall);
+      node->setSymbolReference(d2l);
+      TR_OutlinedInstructions *outlinedHelperCall = new (cg->trHeapMemory()) TR_OutlinedInstructions(node, TR::lcall, targetRegister, CallLabel, reStartLabel, cg);
+      cg->getOutlinedInstructionsList().push_front(outlinedHelperCall);
+
+      cg->decReferenceCount(child);
+      node->setRegister(targetRegister);
+
+      return targetRegister;
+      }
+   else
+      {
+      TR::Register *accReg    = NULL;
+      TR::Register *lowReg    = cg->allocateRegister(TR_GPR);
+      TR::Register *highReg   = cg->allocateRegister(TR_GPR);
+      TR::Register *doubleReg = cg->evaluate(child);
```

Comment:
Shouldn't this be named `floatReg` on this side of the `else`?  i.e., we're doing a `f2l` conversion here.